src/golem_network/golem_network.ts

```diff
@@ -1,42 +1,39 @@
 import { v4 } from "uuid";
 import { YagnaOptions } from "../executor";
 import { Job } from "../job";
-import { Yagna } from "../utils";
+import { Yagna, YagnaApi } from "../utils";
 import { RunJobOptions } from "../job/job";
 import { GolemError } from "../error/golem-error";
 
+type GolemNetworkConfig = Partial<RunJobOptions> & { yagna?: YagnaOptions };
+
 /**
  * The Golem Network class provides a high-level API for running jobs on the Golem Network.
  */
 export class GolemNetwork {
-  private _yagna: Yagna | null = null;
+  private yagna: Yagna;
+  private api: YagnaApi | null = null;
+
+  private initializedState = false;
 
   private jobs = new Map<string, Job>();
 
   /**
    * @param config - Configuration options that will be passed to all jobs created by this instance.
    */
-  constructor(private readonly config: Partial<RunJobOptions> & { yagna?: YagnaOptions }) {}
-
-  private get yagna() {
-    if (this._yagna === null) {
-      throw new GolemError("GolemNetwork not initialized, please run init() first");
-    }
-    return this._yagna;
+  constructor(private readonly config: GolemNetworkConfig) {
+    this.yagna = new Yagna(this.config.yagna);
   }
 
   public isInitialized() {
-    return this._yagna !== null;
+    return this.initializedState;
   }
 
   public async init() {
-    if (this._yagna !== null) {
-      return;
-    }
-    const yagna = new Yagna(this.config.yagna);
-    // this will throw an error if yagna is not running
-    await yagna.connect();
-    this._yagna = yagna;
+    await this.yagna.connect();
+    this.api = this.yagna.getApi();
+
+    this.initializedState = true;
```

Comment:
Not sure we need  all three `api`, `yagna` and `initializedState`, can't we achieve the same business logic with just `api`?

```ts
public async init() {
    if (this.isInitialized()) {
      return;
    }
    const yagna = new Yagna(this.config.yagna);
    // this will throw an error if yagna is not running
    await yagna.connect();
    this.api = this.yagna.getApi();
}

public isInitialized() {
  return this.api !== null
}
```